pkg/daemon/certs/certs.go

```diff
@@ -26,26 +26,63 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"reflect"
 	"syscall"
 	"unsafe"
 
-	"golang.org/x/sys/windows"
+	syswindows "golang.org/x/sys/windows"
 	"k8s.io/klog/v2"
 )
 
-// Reconcile ensures the certificates in the given trusted CA bundle file are imported as system certificates.
+// previousCABundleFileName tracks all the certs that the operator has imported into the node's local trust store
+const previousCABundleFileName = "imported-certs.pem"
+
+// Reconcile ensures the node's system certificates match the expected state as given by the trusted CA bundle file.
 // Returns a boolean whether any certs needed to be reconciled
-func Reconcile(caBundle string) (bool, error) {
-	if caBundle == "" {
-		// TODO: ensure that all certs that we've added previously are removed https://issues.redhat.com/browse/WINC-688
-		return false, nil
+func Reconcile(caBundlePath string) (bool, error) {
+	var expectedCerts []*x509.Certificate
+	var err error
+	// Read expected certs from CA trust bundle file. Will be empty when a cluster-wide proxy is not in use or removed.
+	if caBundlePath != "" {
+		if expectedCerts, err = readCertsFromFile(caBundlePath); err != nil {
+			return false, fmt.Errorf("Failed to read certificate file %s: %w", caBundlePath, err)
+		}
+		klog.Infof("read %d expected Certs from %s", len(expectedCerts), caBundlePath)
 	}
-	// Read expected certs from CA trust bundle file
-	expectedCerts, err := readCertsFromFile(caBundle)
+
+	// Populate existing certs using the previously imported CA trust bundle.
+	pwd, err := os.Getwd()
 	if err != nil {
-		return false, fmt.Errorf("Failed to read certificate file: %v", err)
+		return false, err
 	}
-	return ensureCertsAreImported(expectedCerts)
+	previousCABundlePath := fmt.Sprintf("%s\\%s", pwd, previousCABundleFileName)
```

Comment:
I could just make this a well known location to WICD instead, since it creates/deletes the file. LMK folks